python/rikai/spark/model.py

```diff
@@ -0,0 +1,116 @@
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+"""Wraps SQL ML functionalities in Python.
+"""
+
+import base64
+import logging
+from pathlib import Path
+from typing import Callable, Dict, Optional, Union
+
+from pyspark.serializers import CloudPickleSerializer
+from pyspark.sql import SparkSession
+from pyspark.sql.types import DataType
+
+__all__ = ["create_model"]
+
+_pickler = CloudPickleSerializer()
+
+
+def create_model(
+    name: str,
+    model_uri: Union[str, Path],
+    schema: Union[str, DataType],
+    flavor: Optional[str] = None,
+    preprocessor: Union[str, Callable] = None,
+    postprocessor: Optional[Union[str, Callable]] = None,
+    options: Optional[Dict] = None,
+    replace_if_exist: bool = False,
+):
+    """Create a model.
+
+    This method is equivalent to "CREATE MODEL" SQL statement.
+
+    See Rikai SQL ML reference for details
+
+    Parameters
+    ----------
+    name : str
+        Model name
+    model_uri : str or Path
+        The URI or Path to the serialized Model.
+    schema : str
+        Schema string of the return type of the models.
+    flavor : str
+        Model flavor, can be pytorch, sklearn
+    preprocessor : str or Callback
+        The module string of a preprocessor function,
+        or the post-processor function itself.
+    postprocessor : str or Callback
+        The module string of a post-processor function,
+        or the post-processor function itself.
+    options: Dict[str, str]
+        Additional runtime configuration
+    replace_if_exist : bool
+        Set to true to replace model with the same name.
+
+    Notes
+    -----
+    Please consider this a developer API for model transform developers.
+    End users should not use this method.
+
+    """
+    active_session = SparkSession.getActiveSession()
+    assert (
+        active_session is not None
+    ), "Must run create_model with an active SparkSession"
+
+    logging.info("Register model=%s uri=%s schema=%s", name, model_uri, schema)
+
+    if options is None:
+        options = {}
+    options = {str(k): str(v) for k, v in options.items()}
+
+    preprocessor_bytes = None
+    if isinstance(preprocessor, Callable):
+        preprocessor_bytes = base64.b64encode(
+            _pickler.dumps(preprocessor)
+        ).decode("utf-8")
+        preprocessor = None
+
+    postprocessor_bytes = None
+    if isinstance(postprocessor, Callable):
+        postprocessor_bytes = base64.b64encode(
+            _pickler.dumps(postprocessor)
+        ).decode("utf-8")
+        postprocessor = None
+
+    jvm = active_session._jvm
+    jvm.ai.eto.rikai.sql.spark.parser.RikaiExtSqlParser.initRegistry(
+        jvm.SparkSession.getActiveSession().get()
+    )
+    cmd = jvm.ai.eto.rikai.sql.spark.execution.CreateModelCommand(
+        name,
+        str(model_uri),
+        schema,
+        flavor,
+        preprocessor,
```

Comment:
would it make sense to abstract some of the interface details behind a ModelProcessor class which could have a name and func bytes?

Reply:
or if we can use bytes directly then the datatype would be able to tell us which one it is?

Reply:
this is calling a Scala / Java code.

The type is already explicitly in java code https://github.com/eto-ai/rikai/pull/474/files#diff-a3ca47dbaa1d8ec4cc5fcf62a11aa72ee3cc47a3d564a213fb4ea41b32ab8234R47.

I can change this field to byte[], and do the base64 encoding when passing spec from jvm into python again. 
I did not do it was because 

```
case class ModelSpec
```
currently can be easily decoded via `circe-json`. If these two fields are bytes in JVM, we need to do base64 right before `circe-json` encoding.